nil/contracts/solidity/system/MessageQueue.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import "../lib/IMessageQueue.sol";
+
+struct Message {
+    bytes data;
+    address sender;
```

Comment:
it may be convenient to introduce more fields here (e.g. tokens)

Reply:
In the first version I'd just want to check that it will work at all. basic test passed. We can discussed it and extends current implementation.